pkg/tpath/tpath.go

```diff
@@ -84,6 +85,26 @@ func getPathContext(nc *PathContext, fullPath, remainPath util.Path, createMissi
 	}
 	ncNode := v.Interface()
 
+	// Hack for translation of gateway node
+	if strings.HasSuffix(pe, "gressGateway") {
```

Comment:
This is probably not the best idea - how about turning the below code into a function and calling it with 
handleGateway("ingressGateway")
handleGateway("egressGateway"), then "ILBGateway" etc.